app/controllers/bookings_controller.rb

```diff
@@ -25,7 +25,8 @@ def create
     tickets = []
     ticket_params.each do |ticket_type_id, quantity|
       quantity.to_i.times do
-        user = UserTicket.new(ticket_type_id: ticket_type_id, booking: @booking)
+        user = UserTicket.new(ticket_type_id: ticket_type_id, booking: @booking,
+                              event_source: source_params)
         tickets << user
       end
```

Comment:
There is way too much going on in this controller action which can be moved to the model... ( lean controllers fat models )